sdk/identity/azure-identity/src/main/java/com/azure/identity/IdentityClient.java

```diff
@@ -173,15 +191,37 @@ public IdentityClient(IdentityClientOptions options) {
      */
     public Mono<AccessToken> authenticateWithUsernamePassword(String tenantId, String clientId, String[] scopes, String username, String password) {
         String authorityUrl = options.authorityHost().replaceAll("/+$", "") + "/" + tenantId;
-        ExecutorService executor = Executors.newSingleThreadExecutor();
         try {
-            PublicClientApplication application = PublicClientApplication.builder(clientId)
-                .authority(authorityUrl)
-                .executorService(executor)
-                .build();
+            PublicClientApplication.Builder applicationBuilder = PublicClientApplication.builder(clientId).authority(authorityUrl);
+            if (options.proxyOptions() != null) {
+                applicationBuilder.proxy(proxyOptionsToJavaNetProxy(options.proxyOptions()));
+            }
+            PublicClientApplication application = applicationBuilder.build();
             return Mono.fromFuture(application.acquireToken(UserNamePasswordParameters.builder(new HashSet<>(Arrays.asList(scopes)), username, password.toCharArray()).build()))
-                .map(ar -> new AccessToken(ar.accessToken(), OffsetDateTime.ofInstant(ar.expiresOnDate().toInstant(), ZoneOffset.UTC)))
-                .doFinally(s -> executor.shutdown());
+                .map(ar -> {
+                    currentAccount.set(ar.account());
+                    currentApplication.set(application);
+                    return new AccessToken(ar.accessToken(), OffsetDateTime.ofInstant(ar.expiresOnDate().toInstant(), ZoneOffset.UTC));
+                });
+        } catch (MalformedURLException e) {
+            return Mono.error(e);
+        }
+    }
+
+    public Mono<AccessToken> authenticateWithCurrentlyLoggedInAccount(String[] scopes) {
+        try {
+            SilentParameters parameters;
+            if (currentApplication.get() != null) {
+                if (currentAccount.get() != null) {
```

Comment:
If currentAccount is null how are you getting a token silently?  What account are you getting a token for in this case?

Reply:
the IAccount argument is optional in MSAL - thus try calling that if the account is not available